### PR TITLE
Bug/memory leak fix some refactoring

### PR DESCRIPTION
**Pull request by rfreitas**
## The problem

`react-dropzone` was creating `ObjectURL`s but it did not manage them. This created a memory leak since we were not revoking them.

## Technical Implementation

The `ObjectURL` is automatically revoked and removed from the `File` object, whenever `react-dropzone` creates it.

The `ObjectURL` is still used to create lossy `Base64` and to display Blob or Files in the confirmation step. However these `ObjectURL` are created only when needed and are removed (revoked) when no longer necessary.

It's a good practice not to return these `ObjectURL` to the user, since he might not be aware that he needs to revoke them too.

https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL

Internally `image` was renamed `Blob` since it can always be considered a `Blob` even when it's a `File` (since it's inherited).

`imageLossy` was renamed to `lossyBase64` since it's more descriptive.

## Api changes

the returned capture had it's `image` renamed to `blob`. Open to discussion though.

## Documentation Change

Added an example taken from IDV demo app on how to upload captures to webserver

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://141-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
wow, the size is super big, something to do with source maps :x @hjheath your size plugin already saving our ass.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://141-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by hjheath** [comment]
Yes the size plugin is good. I had a look at it would be trivial to change the status of the check if the size change exceeded a given threshold

**Comment by rfreitas** [comment]
Travis automatic deployment: https://141-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
@hjheath that would be neat!

File: src/components/Capture/capture.js [hunk 1]
```diff
@@ -85,29 +85,40 @@ class Capture extends Component {
...
   }
 
-  handleBase64 = (lossyBase64, base64) => {
-    const blob = base64toBlob(base64)
-    this.handleImage(lossyBase64, blob, base64)
-  }
-
-  handleImage = (base64ImageLossy, image, base64) => {
-    if (!image) {
+  handleImage = (lossyBase64, blob) => {
```
Comment (hjheath) [inline src/components/Capture/capture.js]: I think we can use a better name than `handleImage`

File: src/components/Capture/capture.js [hunk 2]
```diff
@@ -85,29 +85,40 @@ class Capture extends Component {
...
       console.warn('Cannot handle a null image')
       return;
     }
-    const payload = this.createPayload(image, base64ImageLossy, base64)
-    functionalSwitch(this.props.method, {
-      document: ()=> this.handleDocument(payload),
-      face: ()=> this.handleFace(payload)
-    })
+
+    const handleImageInternal = (lossyBase64, blob, base64) => {
```
Comment (hjheath) [inline src/components/Capture/capture.js]: The next 20 lines of logic still feels awkward to me. When would we have a lossyBase64 and a base64 argument? If the answer is never then can't this function just have two arguments (blob, base64)?

File: src/components/Capture/capture.js (hunk 1 above)
Comment (rfreitas) [inline src/components/Capture/capture.js]: suggestions?

File: webpack.config.babel.js [hunk 3]
```diff
@@ -42,6 +42,7 @@ const basePlugins = [
 if (ENV === 'production') {
   basePlugins.push(
     new webpack.optimize.UglifyJsPlugin({
+      sourceMap: true,
```
Comment (hjheath) [inline webpack.config.babel.js]: what does this do?

File: src/components/Capture/capture.js (hunk 2 above)
Comment (rfreitas) [inline src/components/Capture/capture.js]: answer is never, only thing is that to me mixing a lossybase64 and a raw base64 can lead to confusion. They are different base64. Although I agree the whole thing probably needs a rethinking.

File: README.md [hunk 4]
```diff
@@ -263,17 +263,49 @@ Onfido.init({
...
   }
 })
 
-function sendToServer(data) {
-  var request = new XMLHttpRequest()
-  request.open('POST', '/your/server/endpoint', true)
-  request.setRequestHeader('Content-Type', 'application/json')
-  var dataString = JSON.stringify(data)
-  request.send(dataString)
+const reduceObj = (object, callback, initialValue) =>
```
Comment (hjheath) [inline README.md]: These examples look valuable, I would have put them in a separate PR.
1) Do you think they should be part of the SDK code (i.e. we expose `postData` to them).
2) I think we should do the wider rewrite of docs asap.

File: src/components/Capture/capture.js (hunk 2 above)
Comment (rfreitas) [inline src/components/Capture/capture.js]: thinking more about this and you are right, I'll simplify this.

File: webpack.config.babel.js (hunk 3 above)
Comment (rfreitas) [inline webpack.config.babel.js]: makes sure the minifying plugin does not mess with the source maps. But I still haven't completely solved the issue, I'm still not getting perfect source maps on Chrome.

File: README.md (hunk 4 above)
Comment (rfreitas) [inline README.md]: fair point on the separate PR.
Making it part of the SDK would not be a bad idea. What do you think @stefaniacardenas?

File: src/components/Capture/capture.js (hunk 1 above)
Comment (hjheath) [inline src/components/Capture/capture.js]: Ignore, I think `handleImage` is OK. I think `handleImageInternal` can be improved - it does not need to be a function if it's just the implementation of `handleImage`

**Comment by rfreitas** [comment]
Travis automatic deployment: https://141-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://141-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://141-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Capture/capture.js [hunk 5]
```diff
@@ -169,14 +156,18 @@ class Capture extends Component {
       return
     }
 
+    const handleFile = file => fileToBase64(file,
```
Comment (hjheath) [inline src/components/Capture/capture.js]: indents are a bit weird, but otherwise looks good!
